File: go/general/965. 单值二叉树.go
package main

import "fmt"

var res []int

func isUnivalTree(root *TreeNode) bool {
	res := make([]int, 0)
	dfs(root)
	fmt.Println(res)
	if len(res) == 1 {
		return true
	} else {
		return false
	}
}

func dfs(root *TreeNode) {
	if root != nil {
		if !inResult(res, root.Val) {
			res = append(res, root.Val)
		}
		dfs(root.Left)
		dfs(root.Right)
	}
}

func inResult(result []int, k int) bool {
	for _, v := range result {
		if v == k {
			return true
		}
	}
	return false
}

File: python/965. 单值二叉树.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def isUnivalTree(self, root: TreeNode) -> bool:
        res = []
        def dfs(root):
            if root:
                if root.val not in res:
                    res.append(root.val)
                dfs(root.left)
                dfs(root.right)
        dfs(root)
        return True if len(res) == 1 else False